src/lib/Builder/NotificationBuilder.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Builder;
+
+use Twig\Environment;
+
+/**
+ * Builder for notification messages, returns output from twig.
+ */
+class NotificationBuilder
+{
+    const TEMPLATE = 'EzSystemsHybridPlatformUiBundle::notification.html.twig';
+
+    const DEFAULT_TIMEOUT = 10;
+
+    const TYPE_SUCCESS = 'positive';
+
+    const TYPE_ERROR = 'error';
+
+    /**
+     * @var Environment
+     */
+    private $environment;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $message;
+
+    /**
+     * @var int
+     */
+    private $timeout = self::DEFAULT_TIMEOUT;
+
+    /**
+     * @var bool
+     */
+    private $copyable;
+
+    /**
+     * @var string
+     */
+    private $details;
+
+    public function __construct(Environment $environment)
+    {
+        $this->environment = $environment;
+    }
+
+    /**
+     * Sets notification type as success.
+     *
+     * @return $this
+     */
+    public function setSuccess()
+    {
+        $this->type = self::TYPE_SUCCESS;
+
+        return $this;
+    }
+
+    /**
+     * Sets notification type as error.
+     *
+     * @return $this
+     */
+    public function setError()
+    {
+        $this->type = self::TYPE_ERROR;
+
+        return $this;
+    }
+
+    /**
+     * Sets error details.
+     *
+     * @param string $details
+     *
+     * @return $this
+     */
+    public function setErrorDetails(string $details)
+    {
+        $this->copyable = true;
+        $this->details = $details;
+
+        return $this;
+    }
+
+    /**
+     * Sets notification message.
+     *
+     * @param string $message
+     *
+     * @return $this
+     */
+    public function setMessage(string $message)
+    {
+        $this->message = $message;
+
+        return $this;
+    }
+
+    /**
+     * Sets notification timeout.
+     *
+     * @param int $timeout
+     *
+     * @return $this
+     */
+    public function setTimeout(int $timeout)
+    {
+        $this->timeout = $timeout;
+
+        return $this;
+    }
+
+    /**
+     * Builds notification message.
+     *
+     * @return string
+     */
+    public function getResult()
```

Comment:
Anyway, I'm overall not convinced.

The builder doing the rendering doesn't work very well, as explained by Damien and myself. HTML rendering of a notification is only one aspect of it.

In addition, we need to remember that notifications are _mostly_ meant to be passed to the flashbag, so that the app can pick them up during rendering. The builder and the factory are meant to create the object, but do not handle the flashbag. It means that developers using the feature will have to inject both services to create a notification. I'd prefer if we had one service that is used to _add the notification_.